sponsorship_compassion/model/contracts.py

```diff
@@ -581,83 +540,76 @@ def contract_waiting(self, cr, uid, ids, context=None):
                 for line in contract.contract_line_ids:
                     sponsorship = line.sponsorship_id
                     if sponsorship.state == 'active':
-                        cr.execute(
+                        self.env.cr.execute(
                             "update recurring_contract set "
                             "activation_date = current_date,is_active = True "
                             "where id = %s", [contract.id])
             elif contract.type == 'SC':
                 # Activate directly correspondence sponsorships
-                cr.execute(
+                self.env.cr.execute(

```

Comment:
From new API guidelines, I read we should call `self.env.invalidate_all()` after doing queries with the cursor in order to update then environment.